centml/compiler/backend.py

```diff
@@ -76,14 +80,23 @@ def _download_model(self, model_id: str) -> CompiledGraph:
         download_path = get_backend_compiled_forward_path(model_id)
         with open(download_path, "wb") as f:
             f.write(download_response.content)
-            return pickle.loads(download_response.content)
+        return torch.load(download_path)
 
     def _compile_model(self, model_id: str):
-        compile_response = requests.post(
-            url=f"{config_instance.SERVER_URL}/submit/{model_id}",
-            files={"model": pickle.dumps(self.module), "inputs": pickle.dumps(self.inputs)},
-            timeout=config_instance.TIMEOUT,
-        )
+        # The model should have been saved using torch.save when we found the model_id
+        if not os.path.isfile(self.serialized_model_path):
+            raise Exception(f"Model not saved at path {self.serialized_model_path}")
+
+        # Serialize inputs to disk
+        torch.save(self.inputs, self.serialized_input_path, pickle_protocol=config_instance.PICKLE_PROTOCOL)
```

Comment:
can we serialize, save (both module and inputs), and compute model_id in a single function? These operations are related and only needs to be done once. You can either do it in the constructor. Or create an _init() function, And make sure this function is only called once.

Reply:
Right now, we need to be careful that _get_model_id() is only called once. Might be error prone.

Reply:
It might also be a good idea to have clean up function as well. Something to think about.

Reply:
done. moved to a _init() like function